autoextract/aio/client.py

```diff
@@ -34,6 +35,104 @@ class Result(List[Dict]):
+        self._reset()
+        self.pending_queries = query_as_dict_list(query)
+        self._handle_retries = handle_retries
+        self._query_successes = list()
+
+    def _reset(self):
+        """Clear temporary variables between retries"""
+        self.pending_queries = list()
+        self._query_errors = list()
+        self._query_exceptions = list()
+
+    def _process_error(self, query_result):
+        """Process Query-level error.
+
+        Return True if error is retriable or False otherwise.
+        """
+        query_exception = QueryError(
+            query=query_result["query"],
+            message=query_result["error"]
+        )
+        if not query_exception.retriable:
+            return False
+
+        self._query_exceptions.append(query_exception)
+        user_query = query_result["query"]["userQuery"]
+        self.pending_queries.append(user_query)
+        self._query_errors.append(query_result)
+        return True
+
+    def _get_exception_with_longest_timeout(self):
+        """Get the QueryError exception with the longest timeout.
+
+        If an exception doesn't have a timeout (retry seconds),
+        consider it as being zero and prioritize the ones that
+        have retry seconds defined.
+        """
+        return max(
+            self._query_exceptions,
+            key=lambda exc: exc.retry_seconds or 0
+        )
+
+    def get_latest_results(self):
+        """Get latest results (errors + successes).
+
+        This method could be used to retrieve results
+        when an exception is raised while processing results.
+        """
+        return self._query_successes + self._query_errors
+
+    def process_results(self, query_results):
+        """Process query results.
+
+        Return successful queries and also failed ones.
+
+        If `self._handle_exceptions` is True,
+        this method might raise a `QueryError` exception.
+
+        If multiple `QueryError` exceptions are parsed,
+        the one with the longest timeout is raised.
+
+        Successful requests are saved in `self._query_successes`
+        and they are kept between executions
+        while failures are saved in `self._query_errors`.
+
+        Queries saved in `self._query_errors` are moved to
+        `self.pending_queries` between executions.
+        You can use the first or the n-th result:
+
+            - You can get all queries successfully responded in the first try.
+            - You can get all queries successfully in the n-th try.
+            - You may stop with a partial number of successful queries.
+        """
+        self._reset()
+        for query_result in query_results:
+            if self._handle_retries and "error" in query_result:
+                if self._process_error(query_result):
+                    continue
+
+            self._query_successes.append(query_result)
+
+        if self._query_exceptions:
+            raise self._get_exception_with_longest_timeout()
```

Comment:
If not used in other places, it could be a variable here.
Probably, would make it a bit clearer

Reply:
> Probably, would make it a bit clearer

That was my intention when I moved this code to a private class-method but it might have got a little bit more complex and the user needs to refer to this method to understand what's happening here.

Done.